codersdk/provisionerdaemons.go

```diff
@@ -40,7 +40,9 @@ type ProvisionerDaemon struct {
 	ID           uuid.UUID         `json:"id" format:"uuid"`
 	CreatedAt    time.Time         `json:"created_at" format:"date-time"`
 	UpdatedAt    sql.NullTime      `json:"updated_at" format:"date-time"`
```

Comment:
Just a note here that `omitempty` isn't actually relevant here. Since `sql.NullTime` is a struct, it can't be omitted even when the struct is "zero".

https://go.dev/play/p/SGmN64SRt8F

Omitempty _can_ be relevant in e.g. codersdk when we use `codersdk.NullTime` that implements `MarshalJSON`. In that case it might help with generation of TypeScript types (unless we handle the type specifically in gen).

Reply:
@mafredri sounds like `UpdatedAt` should be a `codersdk.NullTime` then in the API struct?

Reply:
Yeah @johnstcn, I think that would be better 👍🏻

Reply:
github.com/coder/coder/pull/11037